index.d.ts

```diff
@@ -3,7 +3,7 @@ import { Action, ActionCreator, AnyAction, StoreEnhancer, Store } from 'redux';
 export interface StoreCreator {
   <S, A extends Action>(
     reducer: LoopReducer<S, A>,
-    preloadedState: S,
+    preloadedState: S | undefined,
```

Comment:
good catch! Thanks! can this be

`preloadedState?: S`

instead?

Reply:
I did try that at first but it doesn't work because the question mark makes it an optional argument--rather than one defined explicitly with an undefined value--and you can't follow an optional argument by a required argument:

```
index.d.ts:7:5 - error TS1016: A required parameter cannot follow an optional parameter.

7     enhancer: StoreEnhancer<S>
      ~~~~~~~~
```

However, I could make them both optional which would follow the [`redux` definition](https://github.com/reduxjs/redux/blob/d53364c44b2fb75b59e2c98090b253c103d63c75/index.d.ts#L232-L236) more closely. Should I define it like this?

```diff
diff --git a/index.d.ts b/index.d.ts
index fa84b3e..9b560f2 100644
--- a/index.d.ts
+++ b/index.d.ts
@@ -3,8 +3,8 @@ import { Action, ActionCreator, AnyAction, StoreEnhancer, Store } from 'redux';
 export interface StoreCreator {
   <S, A extends Action>(
     reducer: LoopReducer<S, A>,
-    preloadedState: S | undefined,
-    enhancer: StoreEnhancer<S>
+    preloadedState?: S,
+    enhancer?: StoreEnhancer<S>
   ): Store<S>;
 }
 
```

The reason I didn't go that way is because the `install()` enhancer is required to make it all work.

Reply:
got it. that makes sense. thanks for explaining